extensions/interactions/DragAndDropSort/directives/DragAndDropSort.js

```diff
@@ -0,0 +1,107 @@
+  function(
+      HtmlEscaperService, dragAndDropSortRulesService,
+      UrlInterpolationService) {
+    return {
+      restrict: 'E',
+      scope: {
+        onSubmit: '&',
+        // This should be called whenever the answer changes.
+        setAnswerValidity: '&'
+      },
+      templateUrl: UrlInterpolationService.getExtensionResourceUrl(
+        '/interactions/DragAndDropSort/directives/' +
+        'drag_and_drop_sort_interaction_directive.html'),
+      controller: [
+        '$scope', '$attrs', 'UrlService', 'EVENT_PROGRESS_NAV_SUBMITTED',
+        function(
+            $scope, $attrs, UrlService, EVENT_PROGRESS_NAV_SUBMITTED) {
+          $scope.choices = HtmlEscaperService.escapedJsonToObj(
+            $attrs.choicesWithValue);
+
+          var answers = [];
+          $scope.list = [];
+          // Make list of dicts from the list of choices.
+          for (var i = 0; i < $scope.choices.length; i++) {
+            $scope.list.push({title: $scope.choices[i], items: []});
+          }
+
+          $scope.submitAnswer = function() {
+            // Converting list of dicts to list of lists to make it consistent
+            // with the ListOfSetsOfHtmlStrings object.
+            for (var i = 0; i < $scope.list.length; i++) {
+              answers.push([$scope.list[i].title]);
+              for (var j = 0; j < $scope.list[i].items; j++) {
+                answers[i].push($scope.list[i].items[j].title);
+              }
+            }
+
+            $scope.onSubmit({
+              answer: answers,
+              rulesService: dragAndDropSortRulesService
+            });
+          };
+
+          $scope.$on(EVENT_PROGRESS_NAV_SUBMITTED, $scope.submitAnswer);
+        }
+      ]
+    };
+  }
+]);
+
+oppia.directive('oppiaResponseDragAndDropSort', [
+  'HtmlEscaperService', 'UrlInterpolationService',
+  function(HtmlEscaperService, UrlInterpolationService) {
+    return {
+      restrict: 'E',
+      scope: {},
+      templateUrl: UrlInterpolationService.getExtensionResourceUrl(
+        '/interactions/DragAndDropSort/directives/' +
+        'drag_and_drop_sort_response_directive.html'),
+      controller: ['$scope', '$attrs', function($scope, $attrs) {
+        $scope.answer = HtmlEscaperService.escapedJsonToObj($attrs.answer);
+      }]
+    };
+  }
+]);
+
+oppia.directive('oppiaShortResponseDragAndDropSort', [
+  'HtmlEscaperService', 'UrlInterpolationService',
+  function(HtmlEscaperService, UrlInterpolationService) {
+    return {
+      restrict: 'E',
+      scope: {},
+      templateUrl: UrlInterpolationService.getExtensionResourceUrl(
+        '/interactions/DragAndDropSort/directives/' +
+        'drag_and_drop_sort_short_response_directive.html'),
+      controller: ['$scope', '$attrs', function($scope, $attrs) {
+        $scope.answer = HtmlEscaperService.escapedJsonToObj($attrs.answer);
+      }]
+    };
+  }
+]);
+
+// Rules Service for DragAndDropSort interaction. Will be implemented in M2.3.
```

Comment:
Remove the second sentence in the comment. You might want to prefix the first sentence with `TODO(vibhor):`